infinitude

```diff
@@ -143,8 +143,14 @@ any '/api/:zone_id/hold' => sub {
 	my $c = shift;
 	my $xml = XML::Simple::Minded->new($store->get('systems.xml'));
 
-	my $idx = ($c->stash('zone_id')||1) - 1;
-	my $zone = $xml->system->config->zones->zone->[$idx];
+	my $zone;
+	my $idx = $c->stash('zone_id')||1;
```

Comment:
I have no idea why I made this 1-based rather than 0-based originally. There may well be a reason that you ran into, but if not, maybe this could be refactored a bit to something like:

```perl
my $idx = $c->stash('zone_id') // 0;
my $zone = $idx eq 'wholeHouse' ? $xml->system->config->wholeHouse
                                : $xml->system->config->zones->zone->[$idx];
```

Reply:
Wouldn't that change the API contract though?

Reply:
ah, derp. 🤦  yeah that's why I did it - everybody wants to call their first zone `1`